test/integration/ae_mdw_web/controllers/tx_controller_test.exs

```diff
@@ -979,17 +972,20 @@ defmodule Integration.AeMdwWeb.TxControllerTest do
     test "get transactions with direction=forward and given GA contract ID with default limit", %{
       conn: conn
     } do
-      criteria = "contract"
       # Generalized Account contract_id
+      limit = 1
       contract_id = "ct_Be5LcGEN2SgZh2kSvf3LqZuawN94kn77iNy5off5UfgzbiNv4"
 
-      conn = request_txs(conn, "forward", criteria, contract_id)
-      response = json_response(conn, 200)
+      assert %{"data" => txs, "next" => next} =
+               conn
+               |> get("/txs/forward", contract: contract_id, limit: limit)
+               |> json_response(200)
+
+      assert ^limit = length(txs)
 
-      Enum.each(response["data"], fn block_tx ->
-        assert block_tx["tx"]["contract_id"] == contract_id
-        assert block_tx["tx"]["type"] == "GAAttachTx"
-      end)
+      assert Enum.all?(txs, fn %{"tx" => %{"contract_id" => c_id, "type" => type}} ->
```

Comment:
Asserting with `Enum.each` gives more detail to the reason of the failture but `Enum.all?` would work fine too.